### PR TITLE
Fix dy calculation in TRD TrackletTransformer

### PR DESCRIPTION
**Pull request by jbarrella**
This PR fixes the calculation and calibration of the deflection length in `TrackletTransformer`. The previous calculations anticipated pre-calibrated incoming tracklets rather than raw ones. With the incoming tracklets being raw, the calculations are much simpler as no reversal of the previous calibration is required. 

@aschmah, @aberdnikova I have three questions that need to be answered to ensure that the new calculation is correct please:
1. What is the distance in local x-direction over which the Lorentz angle is measured? My expectation would be the 3cm of the drift region, but I recall that we used to use 3.35cm which was the height of the anode plane. I'm not certain if either of those are incorrect to use, but I know for sure that we need to be consistent with regard to this.
2. What sign convention do we use for the Lorentz angle? Alex, I sent you a picture of fig. 4.17 from Bailhache's 2009 thesis. My question is basically whether this angle as it is drawn in the picture would be a positive or negative Lorentz angle in the new code. (This is probably something that could easily be verified later on with residuals)
3. I recall that the vDrift value that is calculated in the calibrations is the component of the velocity in x-direction. Is this still the case? Just wanted to confirm.

Please also let me know if I can help in clearing up any of these questions. Thanks!

### CODE REVIEW

File: Detectors/TRD/base/src/TrackletTransformer.cxx
Comment (martenole) [inline Detectors/TRD/base/src/TrackletTransformer.cxx]: Please don't yet send the `calibratedDy` but the `rawDy` instead until we actually have proper calibration

Comment (martenole) [reply Detectors/TRD/base/src/TrackletTransformer.cxx]: Or did you already compare calibrated and raw value and check whether they are in sufficient agreement?

Comment (jbarrella) [reply Detectors/TRD/base/src/TrackletTransformer.cxx]: Well I have a Lorentz angle of 8 degrees hardcoded in there which equates to an offset of about -0.4cm over 3cm. I don't really know whether this would be better or worse for your tracking. I think 8 degrees is a reasonable estimate and hopefully the sign is correct...

Comment (martenole) [reply Detectors/TRD/base/src/TrackletTransformer.cxx]: Maybe you could just run a small simulation and actually compare both values for `dY` which you get?

Comment (jbarrella) [reply Detectors/TRD/base/src/TrackletTransformer.cxx]: Yeah, I did a quick test and all rawDy values are just reduced by ~0.42cm

Comment (martenole) [reply Detectors/TRD/base/src/TrackletTransformer.cxx]: Given that dy ranges from -0.9 cm to 0.9 cm that is not a negligible change.. So the tracker would not find any matches if dy is shifted by that much. Only when we have our actual FEE configuration and the TrapSimulator produces the compatible output as the FEE only then we can tune the calibration of the tracklets and use the calibrated values. Until then we have to stick with the raw values. At the moment the TrapSimulator still does some correction to the tracklets

**Comment by jbarrella** [comment]
@aschmah, @aberdnikova just bumping this for your attention. Not a big rush, but would like to merge when possible

**Comment by jbarrella** [comment]
After looking at the calibration code that @aberdnikova shared with me, the answers to questions 1 and 3 seem to be clear and have not changed since I last looked at the code.

1.  It looks like the anode plane height of 3.35 cm is still used. I noticed that Klein fig. 38 seems to measure the Lorentz angle from the cathode plane and Bailhache fig. 4.17 seems to measure from the pad plane. I think the tracklet can only be approximated as a straight line where the velocity is constant which is only in the drift region. I don't think it actually matters, as long as the `TrackletTransformer` is consistent when it applies the correction, but it might be safer to just use 3 cm. We could also check this at a later stage in the residuals maybe.
2. Let's check in the residuals.
3. It seems clear that this is still the case.

**Review by bazinski — DISMISSED** [review]
Approved but as the ci is failing on the full system test which does execute this it should probably pass that before merging. Even though the failure is mem issues of the digitizer in the test, and hence not related.

**Review by bazinski — DISMISSED** [review]

File: Detectors/TRD/base/src/TrackletTransformer.cxx
Comment (martenole) [inline Detectors/TRD/base/src/TrackletTransformer.cxx]: This should be `mXCathode` instead of `mXAnode` I think

**Review by martenole — CHANGES_REQUESTED** [review]

File: Detectors/TRD/base/src/TrackletTransformer.cxx
Comment (jbarrella) [reply Detectors/TRD/base/src/TrackletTransformer.cxx]: I can't recall if you were in the meeting when we talked about this, but we decided to go with the anode height of 3.35 cm for now. This will be consistent with what is used in the calibration code. However, we will likely revisit this issue in the future since there is still disagreement about 3.35 cm being the most sensible value here, but the change needs to happen in calibration code as well.

**Review by tdietel — DISMISSED** [review]
Looks good to me, including the comments on mXAnode vs mXDrift. We can merge this as is, and revisit when we have a better handle on tracklet resolution and residuals.

We should still check the CI issue raised by Sean.

File: Detectors/TRD/base/src/TrackletTransformer.cxx
Comment (martenole) [reply Detectors/TRD/base/src/TrackletTransformer.cxx]: Well, but if you change it here then this will affect the tracking. Couldn't you change this to be a correction factor for `calibratedDy`? For me `rawDy` should be the deflection that is calculated in the TRAP and there the drift length of 3cm is used and not 3.35 cm.

Comment (jbarrella) [reply Detectors/TRD/base/src/TrackletTransformer.cxx]: Should be resolved now

Comment (martenole) [reply Detectors/TRD/base/src/TrackletTransformer.cxx]: Looks good, thanks

**Review by martenole — APPROVED** [review]

**Comment by jbarrella** [comment]
@TimoWilken Hi, can this be merged? I do not think the errors are caused by me.

**Comment by TimoWilken** [comment]
Hi @jbarrella, would it be alright to wait for another fullCI run when the updated copyright notice template is merged into the o2checkcode check? The macOS error is a known problem at the moment, but fullCI should at least get through the builds and unit tests.

**Comment by martenole** [comment]
@davidrohr could you kindly merge this please? fullCI is green and the two errors from macOS and o2 build are not related.
A followup on this PR is needed in any case. And I checked the TRD reco with the changes from here and it is not affected.

**Pull request merged**
